R/FilteredData.R

```diff
@@ -826,15 +827,30 @@ FilteredData <- R6::R6Class( # nolint
             }
           )
 
+          output$teal_filters_count <- renderText({
+            n_filters_active <- reactive({
+              length(unlist(self$get_filter_state(), recursive = FALSE))
+            })
+            # req(n_filters_active > 0L)
+            sprintf("%s filter%s active", n_filters_active(), ifelse(n_filters_active() == 1, "", "s"))
```

Comment:
- So I think it makes more sense to have a private (or public) method n_filters_active() and then here you can just call `private$n_filters_active()`
- I think it's fine to have "0 filters applied" so can remove the commented out code
- Did we use "active" or "applied" in the earlier case? We should be consistent here
- I'm not sure that `length(unlist(self$get_filter_state(), recursive = FALSE))` is the right number in the MAE case. I would much rather do:

```
lapply(self$datanames(), function(dataname) self$get_filtered_dataset(dataname)$n_active_filter_states())
```

Reply:
> * So I think it makes more sense to have a private (or public) method n_active_filter_states() and then here you can just call `private$n_active_filter_states()`

`length(self$get_filter_state())` is used in two places: in a method and in a server function. Shall I replace those with calls to this new method?

> * I think it's fine to have "0 filters applied" so can remove the commented out code

I slightly disagree, I think a clean output looks neater. I suggest we seek a third opinion.

> * Did we use "active" or "applied" in the earlier case? We should be consistent here

👍 

> * I'm not sure that `length(unlist(self$get_filter_state(), recursive = FALSE))` is the right number in the MAE case. I would much rather do:
> 
> ```
> lapply(self$datanames(), function(dataname) self$get_filtered_dataset(dataname)$n_active_filter_states())
> ```

Indeed. Updated.

> we should make sure everything behaves sensible when we click on these two icons as well (if you're not already)

The "Enable filter panel" button works automagically because it clears filters (and stashes them in another private field).